config/initializers/content_security_policy.rb

```diff
@@ -36,10 +36,10 @@ def host_to_url(str)
   if Rails.env.development?
     webpacker_urls = %w(ws http).map { |protocol| "#{protocol}#{Webpacker.dev_server.https? ? 's' : ''}://#{Webpacker.dev_server.host_with_port}" }
 
-    p.connect_src :self, :data, :blob, assets_host, media_host, Rails.configuration.x.streaming_api_base_url, *webpacker_urls
+    p.connect_src :self, :data, :blob, assets_host, media_host, Rails.configuration.x.streaming_api_base_url, Rails.configuration.x.streaming_api_base_url.gsub(/^ws/, 'http'), *webpacker_urls
```

Comment:
According to https://github.com/mastodon/mastodon/blob/main/app/javascript/mastodon/stream.js#L153, EvenSource will only be used if `streaming_api_base_url` does not start with `ws`. 

As this is a parameter set at startup, we should have a similar check here, and either allow Websockets OR EventSource, not both

Reply:
Oh! I had the understanding that it'd be used for clients that don't support websockets. I must've misread the code a while ago and never adjusted my understanding.

we could certainly lock this down further, but the streaming server always supports both, and to change away from that would add complexity, especially in testing as we'd need to test both eventsource and websockets in configuration.

originally I'd only changed this for development, to make it easier to test both.

Reply:
have removed this change from the PR.